### ISSUE TITLE
The obtained IP address is spoofed by the user

### ISSUE THREAD

**Issue by cxlblm** [akrabat/ip-address-middleware#45]
```PHP
$trustedProxies = ['192.168.97.1'];
$app->add(new RKA\Middleware\IpAddress(true, $trustedProxies));
```
When the value of the `X-Forwarded-For` in my HTTP header is `1.1.1.1,203.0.113.195,150.172.238.178`, the expected IP address to obtain is `150.172.238.178` as the user’s IP. However, the IP address actually obtained is `1.1.1.1`, which appears to be randomly spoofed by the user. After reviewing the `getFirstIpAddressFromHeader` method in the code, it seems the issue lies there. According to the description on MDN regarding the `X-Forwarded-For` value, we should retrieve the first non-trusted proxy IP from the right side, rather than simply taking the leftmost value.

Here's another question: if I want to set the REMOTE_ADDR to *, should I use code like this?
```PHP
$trustedProxies = ['*'];
$app->add(new RKA\Middleware\IpAddress(true, $trustedProxies));
```

**Comment by akrabat** [comment]
The interesting thing here is that https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For states that the order is:

```
X-Forwarded-For: <client>, <proxy1>, <proxy2>
```

Maybe we need an option for this?

**Comment by cxlblm** [comment]
> The interesting thing here is that https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For states that the order is:这里有趣的是https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For指出顺序是：
> 
> ```
> X-Forwarded-For: <client>, <proxy1>, <proxy2>
> ```
> 
> Maybe we need an option for this?也许我们需要一个选择？

The order itself is not the issue. What I want to express is that we should not trust the leftmost address because it can be arbitrarily forged by the user. The `X-Forwarded-For` header is retrieved from the HTTP headers, so users can freely add their own fabricated IP addresses. For example, it could look like this: `X-Forwarded-For: <faker1>, <faker2>, <client>, <proxy1>, <proxy2>`. Therefore, we should extract the `<client>` value, as this is obtained from the `remote_addr` in Nginx or other servers and is guaranteed to be reliable.In `<faker1>, <faker2>, <client>, <proxy1>, <proxy2>`, `<proxy1>` and `<proxy2>` are our trusted IPs. Therefore, when retrieving the user's IP, we should take the IP closest to the trusted proxies, which is `<client>`.

**Comment by akrabat** [comment]
Yes. I finally understood that, but forgot to comment. I am working on it!

**Comment by cxlblm** [comment]
> Yes. I finally understood that, but forgot to comment. I am working on it!是的。我终于明白了，但忘了评论。我正在努力！

Thank you for your efforts; I hope Slim continues to get better and better.

**Comment by akrabat** [comment]
Released in 2.5.0.
